pkg/syncer/syncer.go

```diff
@@ -358,15 +450,19 @@ func (c *Controller) getClient(gvr schema.GroupVersionResource, namespace string
 	return nri
 }
 
-func (c *Controller) inSyncerNamespace(objectNamespace string) bool {
+func (c *Controller) inSyncerNamespace(meta metav1.Object) bool {
 	// If there is no value for the syncer namespace then always process the object
 	// This will also handle the cluster scoped objects case for
 	// objectNamespace when this is not set
 	if c.namespace == "" {
 		return false
 	}
-	if c.namespace == objectNamespace {
-		return true
+	var err error
+	objectNamespace := meta.GetNamespace()
+
+	if c.direction == KcpToPhysicalCluster {
+		_, objectNamespace, err = transformForCluster(schema.GroupVersionResource{}, meta)
```

Comment:
this block is odd. The whole func only makes sense in the `KcpToPhysicalCluster` case. Don't make it generic if it isn't. Do the logic where it is called, not deep in no-op funcs.